### DISCUSSION TITLE
platform_python_implementation in pyproject.toml

### DISCUSSION THREAD

**Discussion by rochamatcomp** [pypa/setuptools#4311] [Q&A] [answered]
Hello, everyone!

I want to [configuring setuptools using pyproject.toml files](https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html). However, I don't know what to do with the `platform_python_implementation` present in `setup.py`:

```
extras_require = {
    ':platform_python_implementation == "CPython"': ["PyDispatcher>=2.0.5"],
    ':platform_python_implementation == "PyPy"': ["PyPyDispatcher>=2.1.0"],
}
```
Does anyone know if there is already a way to declare this? Thank you!

**Comment by abravalheri** [comment, marked as answer]
Hi @rochamatcomp, have you tried adding them to `dependencies`?

Something like:

```toml
dependencies = [
    # ...
    'PyDispatcher>=2.0.5; platform_python_implementation == "CPython"',
    'PyPyDispatcher>=2.0.5; platform_python_implementation == "PyPy"',
]
```
(typing from the phone app, so please double-check).

**Reply by rochamatcomp** [reply]
That way it worked perfectly. Thank you @abravalheri!
